backend/tests/unit/dal/Entities/Extensions/ProjectExtensionsTest.cs

```diff
@@ -0,0 +1,390 @@
+using FluentAssertions;
+using Pims.Dal.Entities;
+using Pims.Dal.Helpers.Extensions;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Xunit;
+
+namespace Pims.Dal.Test.Entities
+{
+    [Trait("category", "unit")]
+    [Trait("category", "dal")]
+    [Trait("category", "entities")]
+    [Trait("group", "entity")]
+    [ExcludeFromCodeCoverage]
+    public class ProjectExtensionsTest
+    {
+        #region Tests
+        #region UpdateProjectNumbers ProjectProperty Entity
+        [Fact]
+        public void UpdateProjectNumbers_Land()
+        {
+            // Arrange
+            var parcel = new Parcel
+            {
+                ProjectNumbers = "[\"1\",\"2\",\"3\",\"4\"]"
+            };
+            var projectProperty = new ProjectProperty()
+            {
+                PropertyType = PropertyTypes.Land,
+                Parcel = parcel
+            };
+
+            // Act
+            var result = projectProperty.UpdateProjectNumbers("3");
+
+            // Assert
+            result.ProjectNumbers.Should().Be("[\"1\",\"2\",\"3\",\"4\"]");
+        }
+
+        [Fact]
+        public void UpdateProjectNumbers_Land_NoProjects()
+        {
+            // Arrange
+            var parcel = new Parcel();
+            var projectProperty = new ProjectProperty()
+            {
+                PropertyType = PropertyTypes.Land,
+                Parcel = parcel
+            };
+
+            // Act
+            var result = projectProperty.UpdateProjectNumbers("3");
+
+            // Assert
+            result.ProjectNumbers.Should().Be("[\"3\"]");
+        }
+
+        [Fact]
+        public void UpdateProjectNumbers_NullLand()
+        {
+            // Arrange
+            var projectProperty = new ProjectProperty()
+            {
+                PropertyType = PropertyTypes.Land,
+            };
+
+            // Act
+            // Assert
+            Assert.Throws<InvalidOperationException>(() => projectProperty.UpdateProjectNumbers("3"));
+        }
+
+        [Fact]
+        public void UpdateProjectNumbers_AddLand()
+        {
+            // Arrange
+            var parcel = new Parcel
+            {
+                ProjectNumbers = "[\"1\",\"2\",\"3\",\"4\"]"
+            };
+            var projectProperty = new ProjectProperty()
+            {
+                PropertyType = PropertyTypes.Land,
+                Parcel = parcel
+            };
+
+            // Act
+            var result = projectProperty.UpdateProjectNumbers("5");
+
+            // Assert
+            result.ProjectNumbers.Should().Be("[\"1\",\"2\",\"3\",\"4\",\"5\"]");
+        }
+
+        [Fact]
+        public void UpdateProjectNumbers_Building()
```

Comment:
Curios, why does the land example add a new project number but the building example just overwrites an existing one?